### PR TITLE
Add NAU7802 docs

### PR DESCRIPTION
**Pull request by cujomalainey**
## Description:
Add docs for NAU7802

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#6291

## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/sensor/nau7802.rst [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+      - platform: nau7802
+        name: "NAU7802 Value"
+        gain: 128
+        ldo_voltage: "3.0V"
+        update_interval: 60s
+
+Configuration variables:
+------------------------
+
+- **name** (**Required**, string): The name for the ADC.
```
Comment (jesserockz) [inline components/sensor/nau7802.rst]: We don't add `name` or `id` in entity docs
```suggestion
```

File: components/sensor/nau7802.rst [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    - ``20``
+    - ``40``
+    - ``80``
+    - ``320``
+
+- **offset_calibration** (*Optional*, int): Pre calibrated offset value, range -8388608 to 8388607
+- **gain_calibration** (*Optional*, float): Pre calibrated gain value, range 0 to 511.9999998807907
+
+- **update_interval** (*Optional*, :ref:`config-time`): The interval to check the sensor. Defaults to ``60s``.
+- **id** (*Optional*, :ref:`config-id`): Manually specify the ID used for code generation.
```
Comment (jesserockz) [inline components/sensor/nau7802.rst]:
```suggestion
```

**Review by jesserockz — CHANGES_REQUESTED** [review]
Don't forget to add this sensor to the `/index.rst` file in the appropriate section.

**Review by jesserockz — APPROVED** [review]

**Comment by cujomalainey** [comment]
Thanks for doing the last part, busy week

**Pull request merged**